atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraServersConfigs.java

```diff
@@ -127,9 +128,9 @@ public CqlCapableConfigTuning tuning() {
         }
 
         @Override
-        public final int numberOfHosts() {
-            Preconditions.checkState(thriftHosts().size() == cqlHosts().size());
-            return thriftHosts().size();
+        @Value.Derived
+        public int numberOfHosts() {
+            return Math.max(thriftHosts().size(), cqlHosts().size());
```

Comment:
This is reasonable, but should include a comment as it's not immediately obvious why you chose this